[next] Permalink

okbel opened the pull request:
### Permalink
- Adds Permalink functionality. A user can share a permalink for a comment.

### Side and core tasks
- Added: Popover Component
- Implementing: ClickOutside and ToggleShow Component https://github.com/coralproject/talk/pull/1751 && https://github.com/coralproject/talk/pull/1757
- Implements `forwardRef`s
- Adds missing imports for UI Components
- Adds functionality to the graph to query by a single comment `comment(id: ID!)`

### Tasks done:
https://www.pivotaltracker.com/story/show/159177760
https://www.pivotaltracker.com/story/show/159177728
https://www.pivotaltracker.com/story/show/158266699
https://www.pivotaltracker.com/story/show/159211298
https://www.pivotaltracker.com/story/show/158925820

## Issues and future discussions
- ~~We need a way to `compose` multiple utilities while dealing with components, i.e `withLocalStateContainer` - `withFragmentContainer`~~
- As is right now the `url` from the `asset` returns the url from the tenant, not the exact url. Comments to replace that in the future have been added.
 
## Updates
- Improved accessibility
- Improved styling
- Improved usability
- Integrated with embed
- Added Tests
- Refactored Stream (_App_ and _Queries_)
- Upgraded typescript to ^3.0.0
- Use local state mutation instead of a reload

alinhle committed:
Merge pull request #1731 from coralproject/next-watcher-flags

Watcher --only

alinhle committed:
Use JSDocs comments (#1727)

wyattjoh committed:
Merge branch 'next' into prevent-compile-loop-relay

alinhle committed:
Merge pull request #1726 from coralproject/prevent-compile-loop-relay

[next] Adapt relay watch config

alinhle committed:
[next] Remove nodemon (#1725)

* Remove old nodemon configs

* Remove nodemon

alinhle committed:
[next] Jest implementation for React Components (#1733)

* Make jest testing work with custom path and css modules

* Add first test

* feat: added unit tests to ci

* fix: updated package-lock.json

* Update cssTransform.js

* Update cssTransform.js

* Fix test in ci

alinhle committed:
Adapt files.exclude (#1736)

okbel committed:
Permalink ui

okbel committed:
Permalink ui

okbel committed:
Adding Copy to clipboard functionality

okbel committed:
WIP

okbel committed:
clean request

okbel commented on src/core/client/stream/components/Comment.tsx:
```diff
@@ -25,6 +23,20 @@ const Comment: StatelessComponent<CommentProps> = props => {
         {props.author && props.author.username}
       </Typography>
       <Typography>{props.body}</Typography>
+      <div className={cn("talk-comment-footer")}>
```
We need to talk about which pattern are we going to use for classes @cvle

okbel committed:
wip

okbel committed:
progress

okbel committed:
progress

okbel committed:
work in progress

okbel committed:
wip

okbel committed:
ui functionality

okbel committed:
Translations :/

okbel committed:
wip

okbel committed:
Merge branch 'next' of github.com:coralproject/talk into permalink

okbel committed:
Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (42 commits)
  [next] Support server side jest testing (#1747)
  Update snapshots
  Add comments
  Remove precss
  Move react-responsive to dev deps
  Remove comment
  Mobile first approach
  Support standard css variables, dynamically set spacing-unit
  Add docs
  Fully implement Flex and MatchMedia
  Responsive Components <3
  fix: linting
  fix: adjusted pageInfo
  Remove obsoloe snapshot
  Move jsdom to dev deps
  Mark comments as always returning a value
  Add comment
  Fix unit tests
  Translate, concept for translation and id strings
  Add aria props
  ...

okbel committed:
Adding Attachment and Popover Component

okbel committed:
merge conflicts

okbel committed:
progress

okbel committed:
Any

okbel committed:
Working

okbel committed:
Support for refs

okbel committed:
Ready

okbel committed:
feat: added support for --only flag

Merge pull request #1731 from coralproject/next-watcher-flags

Watcher --only
Use JSDocs comments (#1727)


Merge branch 'next' into prevent-compile-loop-relay
Merge pull request #1726 from coralproject/prevent-compile-loop-relay

[next] Adapt relay watch config
[next] Remove nodemon (#1725)

* Remove old nodemon configs

* Remove nodemon

[next] Jest implementation for React Components (#1733)

* Make jest testing work with custom path and css modules

* Add first test

* feat: added unit tests to ci

* fix: updated package-lock.json

* Update cssTransform.js

* Update cssTransform.js

* Fix test in ci

Adapt files.exclude (#1736)


Permalink ui

Adding Copy to clipboard functionality

WIP

clean request

wip

progress

progress

work in progress

wip

ui functionality

Translations :/

wip

Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (42 commits)
  [next] Support server side jest testing (#1747)
  Update snapshots
  Add comments
  Remove precss
  Move react-responsive to dev deps
  Remove comment
  Mobile first approach
  Support standard css variables, dynamically set spacing-unit
  Add docs
  Fully implement Flex and MatchMedia
  Responsive Components <3
  fix: linting
  fix: adjusted pageInfo
  Remove obsoloe snapshot
  Move jsdom to dev deps
  Mark comments as always returning a value
  Add comment
  Fix unit tests
  Translate, concept for translation and id strings
  Add aria props
  ...

Adding Attachment and Popover Component

merge conflicts

progress

Any

Working

Support for refs

Ready

okbel committed:
Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (101 commits)
  Ready
  Support for refs
  Working
  Any
  progress
  merge conflicts
  Make timeagoFormatter optional
  More colors
  Colors
  Short circuit endless respawn
  fix: new mongo parser
  Remove jest from watcher config, as it doesnt run well inside
  Add jest set
  Apply suggestions
  Move react-timeago to dev
  Upgrade docz
  Cleanup docz scripts
  Support watcher sets
  [next] Support server side jest testing (#1747)
  Make filter only a pure function
  ...

okbel committed:
feat: added support for --only flag

Merge pull request #1731 from coralproject/next-watcher-flags

Watcher --only
Use JSDocs comments (#1727)


Merge branch 'next' into prevent-compile-loop-relay
Merge pull request #1726 from coralproject/prevent-compile-loop-relay

[next] Adapt relay watch config
[next] Remove nodemon (#1725)

* Remove old nodemon configs

* Remove nodemon

[next] Jest implementation for React Components (#1733)

* Make jest testing work with custom path and css modules

* Add first test

* feat: added unit tests to ci

* fix: updated package-lock.json

* Update cssTransform.js

* Update cssTransform.js

* Fix test in ci

Adapt files.exclude (#1736)


Permalink ui

Adding Copy to clipboard functionality

WIP

clean request

wip

progress

progress

work in progress

wip

ui functionality

Translations :/

wip

Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (42 commits)
  [next] Support server side jest testing (#1747)
  Update snapshots
  Add comments
  Remove precss
  Move react-responsive to dev deps
  Remove comment
  Mobile first approach
  Support standard css variables, dynamically set spacing-unit
  Add docs
  Fully implement Flex and MatchMedia
  Responsive Components <3
  fix: linting
  fix: adjusted pageInfo
  Remove obsoloe snapshot
  Move jsdom to dev deps
  Mark comments as always returning a value
  Add comment
  Fix unit tests
  Translate, concept for translation and id strings
  Add aria props
  ...

Adding Attachment and Popover Component

merge conflicts

progress

Any

Working

Support for refs

Ready

Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (101 commits)
  Ready
  Support for refs
  Working
  Any
  progress
  merge conflicts
  Make timeagoFormatter optional
  More colors
  Colors
  Short circuit endless respawn
  fix: new mongo parser
  Remove jest from watcher config, as it doesnt run well inside
  Add jest set
  Apply suggestions
  Move react-timeago to dev
  Upgrade docz
  Cleanup docz scripts
  Support watcher sets
  [next] Support server side jest testing (#1747)
  Make filter only a pure function
  ...

wyattjoh commented on package.json:
```diff
@@ -24,6 +24,8 @@
   "license": "Apache-2.0",
   "dependencies": {
+    "@types/react-copy-to-clipboard": "^4.2.5",
+    "@types/react-tooltip": "^3.3.5",
```
Should be moved to deps.

okbel replied:
👍 on it

okbel committed:
Merge branch 'permalink' of github.com:coralproject/talk into permalink

* 'permalink' of github.com:coralproject/talk: (102 commits)
  feat: added support for --only flag
  Ready
  Support for refs
  Working
  Any
  progress
  merge conflicts
  Make timeagoFormatter optional
  More colors
  Colors
  Short circuit endless respawn
  fix: new mongo parser
  Remove jest from watcher config, as it doesnt run well inside
  Add jest set
  Apply suggestions
  Move react-timeago to dev
  Upgrade docz
  Cleanup docz scripts
  Support watcher sets
  [next] Support server side jest testing (#1747)
  ...

okbel committed:
New Components

okbel committed:
Merge branch 'next' of github.com:coralproject/talk into permalink

* 'next' of github.com:coralproject/talk: (36 commits)
  [next] Implement AriaInfo (#1756)
  e render props for TrapFocus (#1755)
  [ToggleShow] Fix typo and better description (#1753)
  Fix wrong hour pluralization
  Implement button active state
  Styling fixes
  Fix Playground
  Commit snapshots
  Harmonize typography
  Fix Flex margin bug and add examples
  Update package-lock.json
  Use explicit export
  Implement ButtonIcon component
  Remove wrong ref prop
  Implement new button api
  Read env in watchers.ts
  updated tests
  Update ToggleShow.spec.tsx
  ToggleShow Component
  Fix CI thanks to wyatt
  ...

okbel committed:
wip

okbel committed:
Working

okbel committed:
Placement and docs added

okbel committed:
Now body prop accepts a function or an element

okbel committed:
Support for esc char

okbel committed:
Playground changes

okbel committed:
reordering package json

okbel committed:
Server retrieve by single comment

okbel committed:
Progress:

okbel committed:
Working single comment view

okbel committed:
Removing old implementation of attachment

okbel committed:
Tiny refactor

okbel committed:
Merge branch 'next' of github.com:coralproject/talk into permalink

* 'next' of github.com:coralproject/talk:
  [next] ClickOutside Component (#1757)

okbel committed:
Support for ClickOutside

okbel committed:
Removing old implementation of attachment

Tiny refactor

Merge branch 'next' of github.com:coralproject/talk into permalink

* 'next' of github.com:coralproject/talk:
  [next] ClickOutside Component (#1757)

Support for ClickOutside

okbel committed:
Merge remote-tracking branch 'origin/permalink' into permalink

okbel committed:
Missing imports, redirects Back to the Stream, Share Button, Full func

okbel committed:
Creating assetURL in localState

okbel committed:
S

okbel committed:
Creating assetURL in localState

S

okbel committed:
Merge remote-tracking branch 'origin/permalink' into permalink

okbel committed:
Back to the stream if comment not found

okbel committed:
Added missing dependency and test updated

okbel committed:
Added missing dependency and test updated

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.tsx:
```diff
@@ -0,0 +1,73 @@
+              variant="filled"
+              color="primary"
+              onClick={() => {
+                window.location.href = assetURL;
```
If we use a mutation instead and reset the `commentID`, we can avoid reload to whole page.

alinhle replied:
See https://github.com/coralproject/talk/blob/next/src/core/client/stream/mutations/SetNetworkStatusMutation.ts

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.tsx:
```diff
@@ -0,0 +1,73 @@
+          variant="filled"
+          color="primary"
+          onClick={() => {
+            window.location.href = assetURL;
```
See above

alinhle commented:
> We need a way to compose multiple utilities while dealing with components, i.e withLocalStateContainer - withFragmentContainer

You can do that, just chain it. `withLocalStateContainer(...)(withFragmentContainer()(MyComp))`

alinhle commented on src/core/client/stream/components/Comment/Comment.css:
```diff
@@ -0,0 +1,3 @@
+.footer {
+  margin: 6px 0;
```
If you put a `margin-bottom` here then it'll get in the way of layout components like flex doing their work. Avoid putting margins on the borders, but its ok to layout the margins inside, here: the top margin. Also use `spacingUnits` here it looks like from the design that it's 1.5* spacing unit so, so this should do for this line:
`margin-top: calc(1.5 * var(--spacing-unit));`

alinhle commented on src/core/client/stream/components/Permalink/PermalinkPopover.css:
```diff
@@ -0,0 +1,6 @@
+.textField {
+  margin-right: 5px;
```
Our `Flex` component can add the standard margins, we can use the Flex component instead

alinhle commented on src/core/client/stream/components/Permalink/PermalinkPopover.tsx:
```diff
@@ -0,0 +1,63 @@
+    const { copied } = this.state;
+    return (
+      <ClickOutside onClickOutside={toggleVisibility}>
+        <Flex>
```
If you do `<Flex innerGutter="half">` instead, you'll get the margins for free without adding custom css classes :-)

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.tsx:
```diff
@@ -0,0 +1,73 @@
+  let assetURL: string = "";
+  const query = qs.parse(location.search);
+
+  if (!query.assetID && !query.commentID) {
```
You can get these from the local state, there is no issue chaining withLocalStateContainer and withFragmentContainer :-)

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.tsx:
```diff
@@ -0,0 +1,73 @@
+              }}
+            >
+              Back to the Stream
+            </Button>
```
Hmm I think design is a little different, see: 

![image](https://user-images.githubusercontent.com/14221600/43302692-6e720b42-9142-11e8-84f4-c46df43dcae6.png)

The button appears to be above and with a different styling and text

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.css:
```diff
@@ -0,0 +1,10 @@
+.root {
+  width: 100%;
+  max-width: 400px;
+  margin: 20px auto;
```
Most of the time we'll use spacing units for margins :-)

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.css:
```diff
@@ -0,0 +1,10 @@
+
+.comment,
+.commentNotFound {
+  margin: 20px auto;
```
See above

alinhle commented on src/core/client/stream/containers/CommentContainer.spec.tsx:
```diff
@@ -8,6 +8,7 @@ import { CommentContainer } from "./CommentContainer";
 it("renders username and body", () => {
   const props: PropTypesOf<typeof CommentContainer> = {
     data: {
+      id: "comment-ID",
```
For consistency we could use `comment-id`

alinhle replied:
it's not really important though :-P

alinhle commented on src/core/client/stream/containers/PermalinkViewContainer.tsx:
```diff
@@ -0,0 +1,28 @@
+      }
+    }
+  `,
+})(PermalinkViewContainer);
```
You can just chain another `withLocalStateContainer` here to get the other data you need.

alinhle commented on src/core/client/stream/containers/__snapshots__/CommentContainer.spec.tsx.snap:
```diff
@@ -9,5 +9,6 @@ exports[`renders username and body 1`] = `
   }
   body="Woof"
   createdAt="1995-12-17T03:24:00.000Z"
+  id="comment-ID"
```
For consistency we could use `comment-id` here

alinhle replied:
it's not really important though :-P

alinhle commented on src/core/client/stream/local/initLocalState.ts:
```diff
@@ -21,10 +21,23 @@ export default async function initLocalState(environment: Environment) {
+    if (query.commentID) {
+      localRecord.setValue(query.commentID, "commentID");
+    }
+
```
greaaat

alinhle commented on src/core/client/stream/queries/AppQuery.tsx:
```diff
@@ -49,6 +77,8 @@ const enhanced = withLocalStateContainer<Local>(
+      assetURL
     }
   `
 )(AppQuery);
```
We'll refactor this when we have the iframe.

alinhle replied:
I mean this file

alinhle commented on src/core/client/ui/components/ClickOutside/index.ts:
```diff
@@ -0,0 +1 @@
+export { default, ClickOutside } from "./ClickOutside";
```
Don't export the `ClickOutside` class here (keep the `default`), this is a pattern reserved for testing in codebases with higher order components like ours, but not as a API for consumption.

See: 
- https://stackoverflow.com/questions/48515157/how-to-write-proper-jest-tests-with-hoc
- https://medium.com/@7ynk3r/react-testing-done-right-24fdb4ef43d8
- (there are much more resources somewhere in the big wide web)

Just export the `default` export here:

`export { default } from "./ClickOutside";`

alinhle commented on src/core/client/ui/components/ClickOutside/ClickOutside.tsx:
```diff
@@ -1,12 +1,12 @@
 }
 
-class ClickOutside extends React.Component<Props> {
+export class ClickOutside extends React.Component<ClickOutsideProps> {
```
`default` export here is enough, so we don't need to export this here. This is reserved for testing when having higher order components.

See:
- https://stackoverflow.com/questions/48515157/how-to-write-proper-jest-tests-with-hoc
- https://medium.com/@7ynk3r/react-testing-done-right-24fdb4ef43d8
- (there are much more resources somewhere in the big wide web)

alinhle commented on src/core/client/ui/components/Popover/Popover.css:
```diff
@@ -0,0 +1,15 @@
+  border-radius: 1px;
+  color: #222;
+  display: flex;
+  padding: 6px 10px;
```
Use our magic spacing units here :-) it's probably:
`padding: calc(0.5 * var(--spacing-unit)) var(--spacing-unit);`

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+                    ? body
+                    : body({
+                        toggleVisibility: this.toggleVisibility,
+                        forwardRef: props.ref,
```
You can only use `props.ref` once, it has been already used in line `120`. You probably don't need to pass this at all.

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+  | "left-start";
+
+interface InnerProps {
+  body: (props: RenderProps) => any | React.ReactElement<any>;
```
`body` needs different `RenderProps` from `children`, it shouldn't forward the ref because its already used in line `120`

alinhle replied:
This should resolve to `=> React.ReactNode | React.ReactElement<any>`

okbel replied:
But it passes `toggleVisibility` 🤔 so we do need render props here.

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+
+interface InnerProps {
+  body: (props: RenderProps) => any | React.ReactElement<any>;
+  children: (props: RenderProps) => any;
```
Use `React.ReactNode` instead of `any`

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+interface Props {
+  ref: any;
+  style: CSSProperties;
+}
```
Is this interface really necessary? Doesnt `react-popper` come with types?

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+    return (
+      <Manager>
+        <Reference>
+          {(props: Props) =>
```
Is this type really necessary? Doesnt `react-popper` come with types?
Can't you just destructure it? Shouldnt typescript already know the types?

e.g. `{({ref}) =>`

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+          eventsEnabled
+          positionFixed={false}
+        >
+          {(props: Props) =>
```
Same as above, `react-popper` should already have types, you should be able just to destructure or use it.

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+  body: (props: RenderProps) => any | React.ReactElement<any>;
+  children: (props: RenderProps) => any;
+  description?: string;
+  id?: string;
```
Make this required field, so we enforce accessibility

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+interface InnerProps {
+  body: (props: RenderProps) => any | React.ReactElement<any>;
+  children: (props: RenderProps) => any;
+  description?: string;
```
Make this required field, so we enforce accessibility

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+                  className={cn(styles.root, className)}
+                  ref={props.ref}
+                >
+                  {isValidElement(body)
```
I think `isValidElement` doesn't cover all the valid types for allowed react childrens and is very limiting. Maybe check for `typeof body === 'function'` to check for render  props instead?

okbel replied:
Let's have a talk over it next time :) I can change it for now!

alinhle commented on src/core/client/ui/components/TextField/TextField.css:
```diff
@@ -0,0 +1,11 @@
+  border: 1px solid #979797;
+  box-sizing: border-box;
+  border-radius: 1px;
+  padding: 5px 20px;
```
We should move typography to well typography.css, but you can just add a TODO instead and we'll add it once we have our first real form :-)

alinhle commented on src/core/client/ui/components/TextField/TextField.d.ts:
```diff
@@ -0,0 +1 @@
+export const root: string;
```
This file looks commited by accident

alinhle commented on src/core/client/ui/components/TextField/TextField.tsx:
```diff
@@ -0,0 +1,14 @@
+import * as styles from "./TextField.css";
+
+interface InnerProps extends InputHTMLAttributes<HTMLInputElement> {
+  classes?: typeof styles;
```
Not used, you can remove this for now.

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+  | "left"
+  | "left-start";
+
+interface InnerProps {
```
Maybe call it PopoverProps, we used InnerProps because of HOCs, we don't have one here :-)

alinhle commented on src/core/client/ui/components/Popover/Popover.tsx:
```diff
@@ -0,0 +1,138 @@
+  };
+
+  public handleEsc = (e: KeyboardEvent) => {
+    if (e.key === "Escape") {
```
Loooove it

okbel replied:
<3

alinhle commented on src/core/client/ui/components/Popover/index.ts:
```diff
@@ -0,0 +1,2 @@
+export * from "./Popover";
```
I realized this pattern doesn't always work and isn't future proof (e.g. when testing suddenly requires other exports and we don't to export them here), let's remove this line and explicitly export what we want.

okbel replied:
Ok, removed. Then let's normalize that across our files bc I think I copied that over 💃

alinhle commented on src/core/client/ui/components/TextField/index.ts:
```diff
@@ -0,0 +1,2 @@
+export * from "./TextField";
```
I realized this pattern doesn't always work and isn't future proof (e.g. when testing suddenly requires other exports and we don't to export them here), let's remove this line and explicitly export what we want.

okbel replied:
Same as above: Ok, removed. Then let's normalize that across our files bc I think I copied that over 💃

alinhle replied:
Yes definitely

alinhle commented on src/core/client/ui/components/TextField/TextField.tsx:
```diff
@@ -0,0 +1,14 @@
+
+import * as styles from "./TextField.css";
+
+interface InnerProps extends InputHTMLAttributes<HTMLInputElement> {
```
Let's call it TextFieldProps, we don't use HOCs for this yet

alinhle commented on src/locales/es/stream.ftl:
```diff
@@ -0,0 +1,3 @@
+share = Compartir
+copy = Copiar
+copied = Copiado
```
Just remove these, we don't need to translate now either, we can later use pontoon

alinhle commented on src/core/client/stream/components/Permalink/Permalink.tsx:
```diff
@@ -0,0 +1,38 @@
+import { Button, ButtonIcon, Popover } from "talk-ui/components";
+import PermalinkPopover from "./PermalinkPopover";
+
+interface InnerProps {
```
Call it `PermalinkProps`? We usually used `InnerProps` when we use HOCs but we don't here.

alinhle commented on src/core/client/stream/components/Permalink/Permalink.tsx:
```diff
@@ -0,0 +1,38 @@
+        body={({ toggleVisibility, forwardRef }) => (
+          <PermalinkPopover
+            permalinkUrl={`${assetURL}&commentID=${commentID}`}
+            forwardRef={forwardRef}
```
It looks like the ref is already consumed inside the `Popover`, so this won't be needed, see review of the `Popover` component.

alinhle commented on src/core/client/stream/components/Permalink/PermalinkPopover.tsx:
```diff
@@ -0,0 +1,63 @@
+import { Button, ClickOutside, Flex, TextField } from "talk-ui/components";
+import * as styles from "./PermalinkPopover.css";
+
+interface InnerProps {
```
Call it `PermalinkPopoverProps`? We usually used `InnerProps` when we use HOCs but we don't here.

alinhle commented on src/core/client/stream/components/Permalink/PermalinkView.tsx:
```diff
@@ -0,0 +1,73 @@
+import CommentContainer from "../../containers/CommentContainer";
+import * as styles from "./PermalinkView.css";
+
+export interface InnerProps {
```
Call it `PermalinkViewProps`? We usually used `InnerProps` when we use HOCs but we don't here.

alinhle commented on src/core/client/stream/containers/CommentContainer.tsx:
```diff
@@ -2,16 +2,19 @@ import React, { StatelessComponent } from "react";
+import Comment from "../components/Comment";
 
-type InnerProps = { data: Data } & Omit<CommentProps, keyof Data>;
+export interface InnerProps {
```
Do we need the `InnerProps` to be exported here?

alinhle reviewed (dismissed):
Just lots of tiny things, and 2 not big ones: 
- Use mutation to reset the `commentID`
- Use local state instead of `location`
This looks awesooome @okbel.

okbel committed:
Changes

okbel committed:
More Changes

okbel committed:
docz

okbel committed:
docz

okbel committed:
Adding the mutation

okbel committed:
Changes

okbel committed:
Changes

okbel committed:
Let`s only cover english and then cover all the languagues with Pontoon later

wyattjoh commented:
- [x] Resolve typescript issue

alinhle committed:
Adapt Icon sizes

alinhle committed:
Improve styling, accessebility, usability and use mutation

alinhle committed:
Merge branch 'next' into permalink

alinhle committed:
Merge branch 'next' into permalink

alinhle committed:
Integrate with embed

alinhle committed:
Update tests

alinhle committed:
Refactor stream + some tests

alinhle committed:
Styling improvements and responsibility

alinhle committed:
Add margin, move elevation to theme, use colors from theme

alinhle committed:
More style

alinhle committed:
Don't show icon on xs screens

alinhle committed:
Tweak mdx files

alinhle committed:
Merge branch 'next' into permalink

alinhle committed:
Merge branch 'next' into permalink

alinhle committed:
Add more integration tests

alinhle committed:
Merge branch 'permalink' of https://github.com/coralproject/talk into permalink

alinhle committed:
Add margin on both sides

alinhle committed:
Remove unused dependency

alinhle committed:
Add and refactor tests

alinhle committed:
Fix types
